### PR TITLE
Implement retry logic for failed searches

### PR DESCRIPTION
**Pull request by akarve**
# Description

Client can now send `retry=N` to endpoint for softer searches.
As N increases we search fewer documents per shard, fall back on simple query string, etc.

### CODE REVIEW

File: lambdas/search/index.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ def lambda_handler(request):
     user_source = request.args.get('_source', [])
     # 0-indexed starting position (for pagination)
     user_from = int(request.args.get('from', 0))
-    terminate_after = None  # see if we can skip os.getenv('MAX_DOCUMENTS_PER_SHARD')
+    user_retry = int(request.args.get('retry', 0))
+    terminate_after = os.environ.get('MAX_DOCUMENTS_PER_SHARD') if user_retry else None
```
Comment (akarve) [inline lambdas/search/index.py:42]:
```suggestion
    terminate_after = int(os.environ.get('MAX_DOCUMENTS_PER_SHARD', 10_000)) if user_retry else None
```

Comment (akarve) [inline lambdas/search/index.py:42]:
```suggestion
    terminate_after = int(os.environ.get('MAX_DOCUMENTS_PER_SHARD', 10_000)) if user_retry else None
```

File: lambdas/search/index.py [hunk 2]
```diff
@@ -68,22 +69,37 @@ def lambda_handler(request):
...
+                        # see enterprise/**/bucket.py for mappings
+                        "fields": my_fields
+                    }
                 }
             }
-        }
+        else:
+            body = {
+                "query": {
+                    "simple_query_string" : {
```
Comment (akarve) [inline lambdas/search/index.py:94]:
```suggestion
                    "simple_query_string": {
```

File: lambdas/search/index.py [hunk 3]
```diff
@@ -166,7 +182,7 @@ def lambda_handler(request):
     )
 
     to_search = f"{user_indexes},{index_overrides}" if index_overrides else user_indexes
-
+    print("BODY>>>>>", user_retry, body)
```
Comment (nl0) [inline lambdas/search/index.py:185]: @akarve forgotten debug output?

**Review by nl0 — APPROVED** [review]
lgtm

**Pull request merged**